zhmccli/_cmd_imageprofile.py

```diff
@@ -0,0 +1,1918 @@
+            'problem-state-cpu-counter-authorization-control',
+        'access-crypto-activity-cpu-counter':
+            'crypto-activity-cpu-counter-authorization-control',
+        'access-extended-cpu-counter':
+            'extended-cpu-counter-authorization-control',
+        'access-coprocessor-cpu-counter':
+            'coprocessor-cpu-counter-authorization-control',
+        'access-diagnostic-sampling':
+            'diagnostic-sampling-authorization-control',
+        'access-basic-cpu-sampling': 'basic-cpu-sampling-authorization-control',
+        'permit-aes-key-import': 'permit-aes-key-import-functions',
+        'permit-des-key-import': 'permit-des-key-import-functions',
+        'permit-ecc-key-import': 'permit-ecc-key-import-functions',
+        'access-global-performance-data':
+            'global-performance-data-authorization-control',
+        'permit-global-iocds': 'io-configuration-authorization-control',
+        'permit-cross-partition-authority':
+            'cross-partition-authority-authorization-control',
+        'permit-bcpii-send-commands': 'security-bcpii-send-commands',
+        'permit-bcpii-receive-commands': 'security-bcpii-receive-commands',
+        'bcpii-receive-partitions': 'security-bcpii-receive-partition-list',
+        'time-offset-direction': 'time-offset-increase-decrease',
+    }
+
+    org_options = original_options(options)
+    properties = options_to_properties(org_options, name_map)
+
+    if org_options['disk-partition-id'] == 'auto':
+        properties['disk-partition-id-automatic'] = True
+        # 'disk-partition-id will be unchanged
+    elif org_options['disk-partition-id'] is not None:
+        properties['disk-partition-id-automatic'] = False
+        properties['disk-partition-id'] = str2int(
+            cmd_ctx, 'disk-partition-id', org_options['disk-partition-id'])
+
+    if org_options['absolute-cp-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-cp-capping')
+        properties['absolute-general-purpose-capping'] = value
+
+    if org_options['absolute-ifl-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-ifl-capping')
+        properties['absolute-ifl-capping'] = value
+
+    if org_options['absolute-zaap-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-zaap-capping')
+        properties['absolute-aap-capping'] = value
+
+    if org_options['absolute-ziip-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-ziip-capping')
+        properties['absolute-ziip-capping'] = value
+
+    if org_options['absolute-icf-capping'] is not None:
+        value = absolute_capping_value(
+            cmd_ctx, org_options, 'absolute-icf-capping')
+        properties['absolute-icf-capping'] = value
+
+    if org_options['zaware-host-name'] == '':
```

Comment:
As said, if a simple helper method isn't a good abstraction, maybe there are better ones? At least, you could put the duplicated segments into a distinct method, so that *that* method can be invoked by the different user facing commands.

Reply:
I found a way to remove the duplication, by moving the whole code for handling special options into one function that is used by create and update.